ctapipe/io/magiceventsource.py

```diff
@@ -0,0 +1,1352 @@
+        self.magic_subarray = SubarrayDescription('MAGIC', self.magic_tel_positions, self.magic_tel_descriptions)
+
+    @staticmethod
+    def is_compatible(file_mask):
+        """
+        This method checks if the specified file mask corresponds
+        to MAGIC data files. The result will be True only if all
+        the files are of ROOT format and contain an 'Events' tree.
+
+        Parameters
+        ----------
+        file_mask: str
+            A file mask to check
+
+        Returns
+        -------
+        bool:
+            True if the masked files are MAGIC data runs, False otherwise.
+
+        """
+
+        is_magic_root_file = True
+
+        file_list = glob.glob(file_mask)
+
+        for file_path in file_list:
+            try:
+                import uproot
+
+                try:
+                    with uproot.open(file_path) as input_data:
+                        if 'Events' not in input_data:
+                            is_magic_root_file = False
+                except ValueError:
+                    # uproot raises ValueError if the file is not a ROOT file
+                    is_magic_root_file = False
+                    pass
+
+            except ImportError:
+                if re.match('.+_m\d_.+root', file_path.lower()) is None:
+                    is_magic_root_file = False
+
+        return is_magic_root_file
+
+    @staticmethod
+    def _get_run_number(file_name):
+        """
+        This internal method extracts the run number from
+        the specified file name.
+
+        Parameters
+        ----------
+        file_name: str
+            A file name to process.
+
+        Returns
+        -------
+        int:
+            A run number of the file.
+        """
+
+        mask = ".*\d+_M\d+_(\d+)\.\d+_.*"
+        parsed_info = re.findall(mask, file_name)
+
+        try:
+            run_number = int(parsed_info[0])
+        except IndexError:
+            raise IndexError('Can not identify the run number of the file {:s}'.format(file_name))
+
+        return run_number
+
+    def _set_active_run(self, run_number):
+        """
+        This internal method sets the run that will be used for data loading.
+
+        Parameters
+        ----------
+        run_number: int
+            The run number to use.
+
+        Returns
+        -------
+
+        """
+
+        input_path = '/'.join(self.input_url.split('/')[:-1])
+        this_run_mask = input_path + '/*{:d}*root'.format(run_number)
+
+        run = dict()
+        run['number'] = run_number
+        run['read_events'] = 0
+        run['data'] = MarsDataRun(run_file_mask=this_run_mask)
+
+        return run
+
+    def _generator(self):
+        """
+        The default event generator. Return the stereo event
+        generator instance.
+
+        Returns
+        -------
+
+        """
+
+        return self._stereo_event_generator()
+
+    def _stereo_event_generator(self):
```

Comment:
If this is intended to be called by users, it should not start with an underscore.

I would propose to call it `iter_stereo_events`

Reply:
OK.